alf/algorithms/muzero_algorithm.py

```diff
@@ -13,190 +13,100 @@
-            reanalyze_batch_size (int|None): the memory usage may be too much for
-                reanalyzing all the data for one training iteration. If so, provide
-                a number for this so that it will analyzing the data in several
-                batches.
-            full_reanalyze (bool): if False, during reanalyze only the first
-                ``num_unroll_steps+1`` steps are calculated using MCTS, and the next
-                 ``reanalyze_td_steps`` are calculated from the model directly.
-                 If True, all are calculated using MCTS.
-            priority_func: the function for calculating priority. If it is a str,
-                ``eval(priority_func)`` will be called first to convert it a ``Callable``.
-                It is called as ``priority_func(loss_info)``, where loss_info is
-                the temporally stacked ``LossInfo`` strucuture returned from
-                ``MCTSModel.calc_loss()``.
-            data_transformer_ctor (None|Callable|list[Callable]): if provided,
-                will used to construct data transformer. Otherwise, the one
-                provided in config will be used.
-            data_augmenter: If provided, will be called to perform data augmentation
-                as ``data_augmenter(observation)`` for training observations.
-            target_update_tau (float): Factor for soft update of the target
-                networks used for reanalyzing.
-            target_update_period (int): Period for soft update of the target
-                networks used for reanalyzing.
             config: The trainer config that will eventually be assigned to
                 ``self._config``.
-            enable_amp: whether to use automatic mixed precision for inference.
-                This usually makes the algorithm run faster. However, the result
-                may be different (mostly likely due to random fluctuation).
-            random_action_after_episode_end: If False, the actions used to predict
-                future states after the end of an episode will be the same as the
-                last action. If True, they will be uniformly sampled.
+            enable_amp: whether to use automatic mixed precision for inference. This
+                usually makes the algorithm run faster. However, the result may be
+                different (mostly likely due to random fluctuation). Note that
+                rollout_step is exempted from using AMP.
             debug_summaries (bool):
             name (str):
 
         """
-        model = model_ctor(
-            observation_spec,
-            action_spec,
-            num_unroll_steps=num_unroll_steps,
-            debug_summaries=debug_summaries)
-        mcts = mcts_algorithm_ctor(
+
+        representation_learner = representation_learner_ctor(
             observation_spec=observation_spec,
             action_spec=action_spec,
+            config=config,
```

Comment:
reward_transformer, reward_spec and enable_amp also need to be passed.

Reply:
Fixed.